Adds revised security disclosure process to the docs.

bmbouter opened the pull request:
I built the docs and inspected them; they built correctly and looked correct.

rbarlow commented on docs/dev-guide/contributing/bugs.rst:
```diff
@@ -49,6 +48,22 @@ necessary, the bug may be commented on requesting more information or
+* Reproduction steps
+
+A member of the Pulp Security Team will confirm the vulnerability, determine its impact, and
+develop a fix.
```
Perhaps a "feel free to submit a patch with your disclosure" or similar would be good. Security researchers often like to include patches.

rbarlow commented:
@bmbouter Thanks for taking this effort on, nice work!

bmbouter commented:
@rbarlow Thanks for the review. I added an invitation for them to submit a patch. I also added a release note for 2.9 because this is awesome which links to the new docs. I tested the link too. Merging!
